scanner/scanner.go

```diff
@@ -129,6 +134,20 @@ func (p *PackageMap) ScanPackage() error {
 			return errors.WithStack(err)
 		}
 	}
+
+	// Exclude complete files, if the code is generated.
+	for _, f := range p.pkg.GoFiles {
+		body, err := os.ReadFile(f)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		if !doNotEditRe.Match(body) {
```

Comment:
hmm, maybe it would be good to explicitly apply this to the first line?  
just to be a bit more strict / thorough and not exclude some frankenstein file with a comment floating around somewhere mid-file?

Reply:
@rubensayshi You are right and I am aware of this specification (I linked it myself in https://github.com/hexdigest/gowrap/issues/20). My issue is, that one of my main use-cases is exactly with `gowrap` and this tool generates a valid comment, but does not (yet) comply with strict requirement for the position, because the comment is inserted after the `package` statement.

If you agree, I can change the detection, such that it complies with the test set you linked, which would be sufficient for me.

Reply:
@rubensayshi I added the `isGenerated` function you linked to in a separate commit.

Reply:
ehm, I suppose it's fine to adjust `isGenerated` to also accept `package` before the comment as an exception?

Reply:
This is already the case in the "original" version of `isGenerated`, see: https://github.com/dave/courtney/blob/0e4312f7a7a9fc244e2fbaac78638050451b2324/scanner/generated_test.go#L25

I assume, this because this code predates the decision in https://github.com/golang/go/issues/41196